src/macros.hpp

```diff
@@ -19,6 +19,12 @@
 
 #define PINOCCHIO_STRING_LITERAL(string) #string
 
+#include <exception>
+
+/// \brief Macro to check an assert-like condition and throw a runtime error exception (with a message) if violated.
+#define PINOCCHIO_ASSERT_THROW_AT_RUNTIME(cond) if (!(cond)) { throw std::runtime_error("Wrong argument size"); }
```

Comment:
You should change the message here, putting something more generic.

Reply:
Yes, sounds good. Do you have a suggestion?

Reply:
I will take it in charge in a future PR if you don't mind, in order to provide helpful feedback to the user.